apps/balancer-tools/src/app/apr/round/(components)/MoreFiltersButton.tsx

```diff
@@ -0,0 +1,266 @@
+"use client";
+
+import { capitalize, networksOnBalancer } from "@bleu-balancer-tools/utils";
+import * as Accordion from "@radix-ui/react-accordion";
+import { MixerHorizontalIcon } from "@radix-ui/react-icons";
+import * as Popover from "@radix-ui/react-popover";
+import { usePathname, useRouter, useSearchParams } from "next/navigation";
+import { ChangeEvent, useCallback, useEffect, useState } from "react";
+
+import {
+  AccordionContent,
+  AccordionItem,
+  AccordionTrigger,
+} from "#/components/Accordion";
+import { Badge } from "#/components/Badge";
+import { Checkbox } from "#/components/Checkbox";
+import { BaseInput } from "#/components/Input";
+
+import { PoolTypeEnum } from "../../(utils)/calculatePoolStats";
+import { INITIAL_MIN_TVL } from "../../(utils)/getFilteredApiUrl";
+
+interface SelectedAttributesType {
+  [key: string]: string | string[] | null;
+  network: string[];
+  types: string[];
+  minTvl: string | null;
+  maxTvl: string | null;
+  minApr: string | null;
+  maxApr: string | null;
+  minVotingShare: string | null;
+  maxVotingShare: string | null;
+}
+
+export function MoreFiltersButton() {
+  const router = useRouter();
+  const searchParams = useSearchParams();
+  const pathname = usePathname();
+
+  const filters = [
+    {
+      name: "network",
+      label: "Network",
+      options: Object.values(networksOnBalancer).map(capitalize),
+    },
+    { name: "types", label: "Pool type", options: Object.values(PoolTypeEnum) },
+  ];
+
+  const [selectedAttributes, setSelectedAttributes] =
+    useState<SelectedAttributesType>({
+      network: [] as string[],
+      types: [] as string[],
+      minTvl: String(INITIAL_MIN_TVL),
+      maxTvl: null,
+      minApr: null,
+      maxApr: null,
+      minVotingShare: null,
+      maxVotingShare: null,
+    });
+
+  useEffect(() => {
+    const valuesAttributes = [
+      "minTvl",
+      "maxTvl",
+      "minApr",
+      "maxApr",
+      "minVotingShare",
+      "maxVotingShare",
+    ];
+    const arrayAttributes = ["network", "types"];
+    const updatedAttributes = { ...selectedAttributes };
+    valuesAttributes.forEach((value) => {
+      const paramValue = searchParams.get(value);
+      if (paramValue !== null) {
+        updatedAttributes[value] = paramValue;
+      }
+    });
+
+    arrayAttributes.forEach((value) => {
+      if (searchParams.has(value)) {
+        updatedAttributes[value] = searchParams.get(value)?.split(",") || [];
+      }
+    });
+    setSelectedAttributes(updatedAttributes);
+  }, []);
+
+  const pushQueryparamsURL = () => {
+    const current = new URLSearchParams(Array.from(searchParams.entries()));
+
+    Object.entries(selectedAttributes).map(([name, values]) => {
```

Comment:
would make sure this is wrapped in a useCallback with `selectedAttributes` in the dep array